### ISSUE TITLE
Jumper cable coming loose causes brief loss of vehicle control

### ISSUE THREAD

**Issue by Regularitee** [CleverRaven/Cataclysm-DDA#25548]
If the player is driving a vehicle connecting to something else by jumper cable, and the jumper cable becomes overextended and breaks loose, the player briefly loses control of the vehicle. It's similar to what you would experience if you hit the "wait" key several times in a row: you cannot steer the vehicle, and instead continue forward at whatever speed cruise control is set at.

**Comment by narc0tiq** [comment]
Yup, I can confirm with a quick code read at <https://github.com/CleverRaven/Cataclysm-DDA/blob/86ff32930b80bd705ce2e1801aebb429a715b130/src/item.cpp#L6173-L6188>
 This should only ever happen while you're carrying the cable yourself: <https://github.com/CleverRaven/Cataclysm-DDA/blob/86ff32930b80bd705ce2e1801aebb429a715b130/src/item.cpp#L6164>

It's consuming 30 or 200 moves every time the cable is reset, regardless of how extended it was (this should nominally translate to about 0.3/2 turns, but depends on character speed). A nicer solution would be to use `max_charges - charges` (saved *before* resetting the item data), though of course for an actually over-extended cable, this won't make much difference.

Someone should definitely double-check that consuming `p->moves` correctly deducts just the 30/200 moves expected.

**Comment by nexusmrsep** [comment]
I wonder it it was intended. If you'd drive a car fixed to a cable it would pull you before breaking and that might make vehicle uncontrollable for a moment, although it's true when it comes to tug cable and not jumper cable.

**Comment by AMurkin** [comment]
Related #17772, #22529.

**Comment by dobbry-vechur** [comment]
The old code path that immediately spent player moves when a carried jumper cable reset is gone. Current overextension handling calls `reset_link()`, which can mark a long cable as `needs_reeling`, but does not immediately consume moves or assign `ACT_REEL_CABLE`. Reeling is now a separate activity when the player later manages/detaches the cable.

Related issues were also fixed. Closing.
